dap_prinz_green_jobs/pipeline/green_measures/industries/sic_mapper/sic_mapper.py

```diff
@@ -0,0 +1,434 @@
+
+        Args:
+            preprocessed_job_adverts (List[Dict[str, str]]): A list of job adverts with pre-processed job descriptions.
+
+        Returns:
+            List[Dict[str, str]]: A list of job adverts with the company description.
+        """
+        logger.info(
+            f"Extracting company descriptions from {len(preprocessed_job_adverts)} job adverts..."
+        )
+        company_descriptions = []
+        for job_advert in tqdm(preprocessed_job_adverts):
+            company_description = ""
+            for sentence in job_advert[f"{self.job_description_key}_sentences"]:
+                if (
+                    10 < len(sentence) < 250
+                ):  # Only predict on reasonable length sentences
+                    pred = self.company_description_classifier(sentence)[0]
+                    if pred["label"] == "LABEL_1":
+                        company_description += f"{sentence}. "
+
+            company_descriptions.append(
+                {
+                    self.job_id_key: job_advert[self.job_id_key],
+                    self.company_name_key: job_advert[self.company_name_key],
+                    f"{self.job_description_key}_clean": job_advert.get(
+                        f"{self.job_description_key}_clean"
+                    ),
+                    "company_description": company_description.strip(),
+                }
+            )
+
+        return company_descriptions
+
+    def get_company_description_embeddings(
+        self, company_descriptions_dict: Dict[str, str]
+    ) -> Dict[str, str]:
+        """Gets company description embeddings.
+
+        Args:
+            company_descriptions_dict (Dict[str, str]): A dictionary
+                of company description hashes and company descriptions.
+        """
+        logger.info(
+            f"Computing embeddings for {len(company_descriptions_dict)} company descriptions..."
+        )
+
+        comp_embeds = self.bert_model.transform(
+            list(company_descriptions_dict.values())
+        )
+        comp_embeds_dict = dict(
+            zip(list(company_descriptions_dict.keys()), comp_embeds)
+        )
+
+        return comp_embeds_dict
+
+    def create_vector_index(self, sic_embeddings: np.ndarray) -> faiss.IndexFlatIP:
+        """Creates a FAISS index for a given set of embeddings.
+
+        Args:
+            embeddings (np.ndarray): Embeddings to put in the index.
+
+        Returns:
+            faiss.IndexFlatIP: A FAISS index.
+        """
+        logger.info(f"Creating FAISS index for vectors...")
+        d = sic_embeddings.shape[1]  # define the dimensionality of the vectors
+        # lets use brute force L2
+        llm_index = faiss.IndexFlatIP(d)
+        faiss.normalize_L2(sic_embeddings)  # normalise to use cosine distance
+        llm_index.add(sic_embeddings)  # add vectors to the index
+
+        return llm_index
+
+    def predict_sic_code(
```

Comment:
I'd consider a refactor where you are allowed to input several company embeddings rather than one at a time. If faiss is anything like other distance algorithms then it will speed things up to do them in (reasonably sized) batches rather than one by one